keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlLock.kt

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.keel.sql
+
+import com.netflix.spinnaker.keel.sync.Lock
+import org.jooq.DSLContext
+import org.jooq.Field
+import org.jooq.Record
+import org.jooq.Table
+import org.jooq.impl.DSL.field
+import org.jooq.impl.DSL.table
+import org.slf4j.LoggerFactory
+import java.sql.Timestamp
+import java.time.Clock
+import java.time.Duration
+
+class SqlLock(
+  private val jooq: DSLContext,
+  private val clock: Clock
+) : Lock {
+  override fun tryAcquire(name: String, duration: Duration): Boolean =
```

Comment:
This is the actual SQL lock impl. If you're skim reading this PR, this is the important bit.